spec/defines/kernel_opt_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'spec_helper'
+
+describe 'grubby::kernel_opt' do
+  let(:title) { 'foo' }
+
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:facts) do
+        os_facts
+      end
+
+      it { is_expected.to compile }
+      context 'with no parameters' do
+        it do
+          is_expected.to contain_exec('Ensure foo kernel option is set for DEFAULT').with(
+            command: '/sbin/grubby --update-kernel=DEFAULT --args=foo',
+            unless: '/sbin/grubby --info=DEFAULT | grep ^args= | test -z "$(grep -wv foo)"',
+          )
+        end
+        context 'with value and scope set' do
+          let(:params) do
+            { value: 'bar', scope: 'ALL' }
+          end
+
+          it do
+            is_expected.to contain_exec('Ensure foo=bar kernel option is set for ALL').with(
+              command: '/sbin/grubby --update-kernel=ALL --args=foo=bar',
+              unless: '/sbin/grubby --info=ALL | grep ^args= | test -z "$(grep -wv foo=bar)"',
+            )
+          end
+        end
+      end
+      context 'with ensure absent parameters' do
+        let(:params) do
+          { ensure: 'absent' }
+        end
+
+        it do
+          is_expected.to contain_exec('Ensure foo kernel option is absent for DEFAULT').with(
+            command: '/sbin/grubby --update-kernel=DEFAULT --remove-args=foo',
+            unless: '/sbin/grubby --info=DEFAULT | grep ^args= | test -z "$(grep -w foo)"',
+          )
+        end
+        context 'with value and scope set' do
+          let(:params) do
+            super().merge(value: 'bar', scope: 'ALL')
+          end
+
+          it do
+            is_expected.to contain_exec('Ensure foo=bar kernel option is absent for ALL').with(
```

Comment:
I imagine here we ensure 'foo' is absent no matter if value is provided or not.
I have not run that in real world, but I think the test becomes:
```suggestion
            is_expected.to contain_exec('Ensure foo kernel option is absent for ALL').with(
```

Reply:
It's a bit tricky since to ensure a foo=bar is absent you have to specify the value as well in the grep since only `foo=bar` is a word (-w to grep) .

Reply:
My mistake:

```
cat << EOF | grep '^args='  | test -z "$(grep -w rhgb)"
args="foo rhgb=dark"
args="bar rhgb=dark"
EOF

echo "Delete rhgb ? Should return 1 since rhgb present in all lines, return code was  $?"
```

and it does return one , guess `=` is considered a word end.